test/pkg/flag/flag.go

```diff
@@ -1,31 +1,105 @@
-// This package augments the default "flags" package with functionality similar
-// to that in "github.com/urfave/cli", since the two packages do not mix easily
-// and the "testing" package uses a default flagset that we cannot easily update.
 package flag
 
 import (
 	"flag"
 	"strings"
 
 	"github.com/sirupsen/logrus"
+	"golang.org/x/exp/maps"
+	"golang.org/x/exp/slices"
 )
 
-const FeatureFlagName = "feature"
+const (
+	FeatureFlagName        = "feature"
+	ExcludeFeatureFlagName = "exclude"
+)
+
+// NewFeatureFlag defines two flags, [FeatureFlagName] and [ExcludeFeatureFlagName], to all
+// setting and excluding certain features.
+func NewFeatureFlag(features []string) *ExcludeStringSet {
+	fs := NewStringSet(FeatureFlagName,
+		"`features` to test; can be set multiple times, with a comma-separated list, or both. "+
+			"Leave empty to enable all features. "+
+			"(supported features: "+strings.Join(features, ", ")+")", false)
+
+	return NewExcludeStringSet(fs, ExcludeFeatureFlagName,
+		"`features` to exclude from tests (see "+FeatureFlagName+" for more details)",
+		features)
+}
+
+// ExcludeStringSet allows unsetting strings seen in a [StringSet].
+type ExcludeStringSet struct {
```

Comment:
>ExcludeStringSet [](http://example.com/codeflow?start=5&length=16)

should this be called something like  IncExclStringSet since it has both?

Reply:
I figured `StringSet` implied inclusion, but I can rename if you think its clearer.

Reply:
ExcludeStringSet makes it sound like it only has the excluded ones.